crates/nargo/src/cli/mod.rs

```diff
@@ -206,6 +207,37 @@ fn path_to_stdlib() -> PathBuf {
     dirs::config_dir().unwrap().join("noir-lang").join("std/src")
 }
 
+// Removes duplicates from the list of public input witnesses
+fn dedup_public_input_indices(indices: PublicInputs) -> PublicInputs {
+    let duplicates_removed: HashSet<_> = indices.0.into_iter().collect();
+    PublicInputs(duplicates_removed.into_iter().collect())
+}
+
+// Removes duplicates from the list of public input witnesses and the
+// associated list of duplicate values.
+pub(crate) fn dedup_public_input_indices_values(
+    indices: PublicInputs,
+    values: Vec<FieldElement>,
+) -> (PublicInputs, Vec<FieldElement>) {
+    // Assume that the public input index lists and the values contain duplicates
+    assert_eq!(indices.0.len(), values.len());
+
+    let mut public_inputs_without_duplicates = Vec::new();
+    let mut already_seen_public_indices = HashSet::new();
+
+    for (index, value) in indices.0.iter().zip(values) {
+        if !already_seen_public_indices.contains(&index) {
```

Comment:
We can create another issue to add an error when the values are not the same. We won't be able to do it in this method as is because it does not contain enough information to give meaningful error messages, like the variable names in the .TOML file which have colliding witness indices.

This may become easier when we have the Abi to witness map PR

Reply:
Yes you can create an issue for this, but we should still assert that the values are the same.
When the issue is fixed, this assert should not raise anymore.

Reply:
Done